File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/rpc/ProductClient.java
package io.micronaut.nats.docs.rpc;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.nats.annotation.NatsClient;
import io.micronaut.nats.annotation.Subject;
import org.reactivestreams.Publisher;
// end::imports[]

@Requires(property = "spec.name", value = "RpcUppercaseSpec")
// tag::clazz[]
@NatsClient
public interface ProductClient {

    @Subject("product")
    String send(String data); // <1>

    @Subject("product")
    Publisher<String> sendReactive(String data); // <2>
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/nats/docs/rpc/ProductClient.kt
package io.micronaut.nats.docs.rpc

import io.micronaut.context.annotation.Requires
// tag::imports[]
import io.micronaut.nats.annotation.NatsClient
import io.micronaut.nats.annotation.Subject
import org.reactivestreams.Publisher
// end::imports[]

@Requires(property = "spec.name", value = "RpcUppercaseSpec")
// tag::clazz[]
@NatsClient
interface ProductClient {

    @Subject("product")
    fun send(data: String): String // <1>

    @Subject("product")
    fun sendReactive(data: String): Publisher<String> // <2>
}
// end::clazz[]
